src/main/java/com/hankcs/algorithm/AhoCorasickDoubleArrayTrie.java

```diff
@@ -114,6 +114,33 @@ public void parseText(String text, IHit<V> processor)
      * @param text The text
      * @param processor A processor which handles the output
      */
+    public void parseText(String text, IHitCancellable<V> processor)
+    {
+        int position = 1;
+        int currentState = 0;
+        for (int i = 0; i < text.length(); ++i)
+        {
+            currentState = getState(currentState, text.charAt(i));
+            int[] hitArray = output[currentState];
+            if (hitArray != null)
+            {
+                for (int hit : hitArray)
+                {
+                    boolean proceed = processor.hit(position - l[hit], position, v[hit]);
+                    if(!proceed) {
+                        return;
+                    }
+                }
+            }
+            ++position;
```

Comment:
`position` is, at every cycle, `i+1`. You can just do `final int position = i + 1;` instead of having it as a global and mutable var.

Reply:
This, in general, is hard to read. Using `streams` would make this cleaner and more functional, but not sure if it's worth it considering the rest of the code (it's always worth it living it better than how you found it, but...)